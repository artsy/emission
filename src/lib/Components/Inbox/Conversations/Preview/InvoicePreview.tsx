import * as React from "react"
import { Image, Text, TouchableHighlight } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import styled from "styled-components/native"

import colors from "../../../../../data/colors"
import fonts from "../../../../../data/fonts"

import SerifText from "../../../Text/Serif"

import InvertedButton from "../../../Buttons/InvertedButton"

const Container = styled.View`
  border-width: 1;
  border-color: ${colors["gray-regular"]};
  flex: 1;
  flex-direction: row;
`

const TextContainer = styled.View`
  flex: 1;
  flex-direction: column;
  margin-left: 25;
  align-self: center;
`

const Icon = styled(Image)`
  resize-mode: contain;
  width: 40;
  margin-top: 12;
  margin-left: 12;
  margin-bottom: 12;
`

const CanceledLabel = styled.Text`
  color: ${colors["gray-medium"]};
  font-weight: bold;
`
const RefundedLabel = styled.Text`
  color: ${colors["gray-medium"]};
  font-weight: bold;
`
const PaidLabel = styled.Text`color: ${colors["green-regular"]};`
const PayButtonContainer = styled.View`
  height: 25;
  width: 71;
`

const PaymentRequest = styled.Text`
  font-family: ${fonts["garamond-regular"]};
  font-size: 14;
`

interface Props extends RelayProps {
  onSelected?: () => void
}

interface InvoiceStateButtonProps {
  state: string | null
}

const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
  if (state === "PAID") {
    return <PaidLabel>PAID</PaidLabel>
  }
  if (state === "VOID") {
    return <CanceledLabel>CANCELED</CanceledLabel>
  }
  if (state === "REFUNDED") {
    return <RefundedLabel>REFUNDED</RefundedLabel>
  }
  if (state === "UNPAID") {
    return (
      <PayButtonContainer>
        <InvertedButton text="PAY" />
      </PayButtonContainer>
    )
  }
}

export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
    <Container>
      <Icon source={require("../../../../../../images/payment_request.png")} />
      <TextContainer>
        <PaymentRequest>Payment request</PaymentRequest>
        <PaymentRequest>
          {invoice.total}
        </PaymentRequest>
      </TextContainer>
      <TextContainer>
        {InvoiceStateButton(invoice)}
      </TextContainer>
    </Container>
  </TouchableHighlight>

export default createFragmentContainer(
  InvoicePreview,
  graphql`
    fragment InvoicePreview_invoice on Invoice {
      payment_url
      state
      total
    }
  `
)

interface RelayProps {
  invoice: {
    payment_url: string | null
    state: string | null
    total: string | null
  }
}
