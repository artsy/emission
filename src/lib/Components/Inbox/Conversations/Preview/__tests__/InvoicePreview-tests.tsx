import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { InvoicePreview } from "../InvoicePreview"

it("renders correctly", () => {
  const tree = renderer.create(<InvoicePreview invoice={invoice} />)
  expect(tree).toMatchSnapshot()
})

const invoice = {
  state: "REFUNDED",
  payment_url: "https://adopt-cats.org/pay-here",
  total: "$420",
}
