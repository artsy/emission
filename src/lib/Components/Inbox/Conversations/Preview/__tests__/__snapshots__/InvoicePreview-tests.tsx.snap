// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoicePreview renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#e5e5e5",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 72,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../images/payment_request.png",
      }
    }
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "resizeMode": "contain",
          "width": 40,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 14,
          },
          undefined,
        ]
      }
    >
      Payment request
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "AGaramondPro-Regular",
            "fontSize": 14,
          },
          Array [
            Object {
              "fontWeight": "bold",
            },
            undefined,
          ],
        ]
      }
    >
      $420
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 25,
          "marginRight": 15,
          "width": 78,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#cccccc",
            "fontFamily": "AvantGardeGothicITC",
            "fontSize": 12,
            "lineHeight": 25,
            "textAlign": "right",
          },
          undefined,
        ]
      }
    >
      REFUNDED
    </Text>
  </View>
</View>
`;
